test/test_nn.py

```diff
@@ -6193,15 +6193,17 @@ def test_batchnorm_cudnn_half(self):
         thnn_output = m(input)
         thnn_output.sum().backward()
         thnn_input_grad = input.grad.data.clone()
-        self.assertEqual(thnn_output.type(), input.type())
+        self.assertEqual(thnn_output.device, input.device)
```

Comment:
These 2 lines are encountered sufficient number of times to merit their own helper function?